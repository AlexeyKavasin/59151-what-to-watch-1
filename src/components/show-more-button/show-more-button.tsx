import * as React from "react";
import {IShowMoreButton, IShowMoreButtonState} from "../../interfaces";

export class ShowMoreButton extends React.Component<IShowMoreButton, IShowMoreButtonState> {
    constructor(props) {
        super(props);
        this.state = {
            filmsShown: 0,
        }
    }

    componentDidMount() {
        const {filmsToShow} = this.props;
        this.setState({
            filmsShown: this.state.filmsShown + filmsToShow
        });
    }

    render() {
        const {filmsLength, filmsToShow, onShowMoreClick} = this.props;
        const allFilmsShown = filmsToShow >= filmsLength;

        return <React.Fragment>
            <div className="catalog__more" style={{display: allFilmsShown ? `none` : `block`}}>
                <button className="catalog__button" type="button" onClick={(evt) => {
                    evt.preventDefault();
                    onShowMoreClick(this.state.filmsShown + filmsToShow);
                }}>Show more</button>
            </div>
        </React.Fragment>
    }
}
