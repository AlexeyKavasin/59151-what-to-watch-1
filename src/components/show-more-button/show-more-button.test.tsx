import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {HashRouter} from 'react-router-dom';
import {ShowMoreButton} from './show-more-button';

it(`Show more button renders correctly`, () => {
  const tree = renderer
    .create(
      <HashRouter>
        <ShowMoreButton
          filmsToShow={20}
          onShowMoreClick={jest.fn()}
          filmsLength={20}
        />
    </HashRouter>).toJSON();

  expect(tree).toMatchSnapshot();
});
