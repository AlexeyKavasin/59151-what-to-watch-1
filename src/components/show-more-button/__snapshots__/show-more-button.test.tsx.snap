// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show more button renders correctly 1`] = `
<div
  className="catalog__more"
  style={
    Object {
      "display": "none",
    }
  }
>
  <button
    className="catalog__button"
    onClick={[Function]}
    type="button"
  >
    Show more
  </button>
</div>
`;
